data/mods/gen6/random-teams.ts

```diff
@@ -785,7 +785,7 @@ export class RandomGen6Teams extends RandomGen7Teams {
 
 			// Choose next 4 moves from learnset/viable moves and add them to moves list:
 			while (moves.length < 4 && movePool.length) {
-				const moveid = this.sampleNoReplace(movePool);
+				const moveid = this.sampleNoReplace(movePool) as ID;
```

Comment:
Why all the casts?

Reply:
`movePool` should be an array of `ID`s already?

Reply:
@KrisXV?

Reply:
Oh, I see. `movePool` can't be an array of IDs because it would require casting on every `.includes()` check. `rejectedPool` should still be declared as an `ID[]`, though; that shouldn't cause any typing problems.